src/placeos-rest-api/controllers/metadata.cr

```diff
@@ -73,17 +74,28 @@ module PlaceOS::Api
       meta = Model::Metadata.for(parent_id, metadata.name).first?
 
       if meta
+        # Check if the current user has access
+        raise Error::Forbidden.new unless is_support? || parent_id == user_token.id || (meta.editors & Set.new(user_token.user.roles)).size > 0
```

Comment:
can `meta.editors` be nil here?

Reply:
the access check is slightly duplicated

maybe a method like so could be good (i know it's a minimal code path but just to ensure it's tested)

```crystal
def self.has_metadata_access(user_token, parent_id, metadata = nil)
  # Does user own the metadata, or are they a support user?
  user_access = user_token.is_support? || parent_id = user_token.id
  if metadata.nil?
    user_access
  else
    # Does the user have access to the metadata?
    user_access  || (meta.editors & Set.new(user_token.user.roles)).size > 0
  end
end
```

Reply:
the two code paths are protecting against different things.

* create - support users
* update - support users and editors

`meta.editors` cannot be nil, also the compiler would have complained :)

Reply:
i guess that's what the method is doing, they could be split into 2 or just inlined - maybe we could name the paths to make it obvious/testable since it's regarding access. also happy to move on